pkg/registration/hub/gc/controller.go

```diff
@@ -0,0 +1,80 @@
+package gc
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	rbacv1client "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	rbacv1listers "k8s.io/client-go/listers/rbac/v1"
+	"k8s.io/client-go/tools/cache"
+
+	informerv1 "open-cluster-management.io/api/client/cluster/informers/externalversions/cluster/v1"
+	worklister "open-cluster-management.io/api/client/work/listers/work/v1"
+
+	"open-cluster-management.io/ocm/pkg/common/queue"
+)
+
+// gcReconcile is an interface for reconcile cleanup logic.
+type gcReconcile interface {
+	reconcile(ctx context.Context, clusterName string) (bool, error)
```

Comment:
we had a change to refactor the gc part. I would think the return should also have a reconcileStop/Continue, so we can know further gc is needed. In the future I think we could build a reconcile chain with metaObjectClient.